ebpf/address.go

```diff
@@ -0,0 +1,95 @@
+package ebpf
+
+import (
+	"encoding/binary"
+	"net"
+
+	"github.com/mailru/easyjson/jwriter"
+)
+
+// Address is an IP abstraction that is family (v4/v6) agnostic
+type Address interface {
+	Bytes() []byte
+	String() string
+	MarshalEasyJSON(w *jwriter.Writer)
+}
+
+type v4Address [4]byte
+
+// V4Address creates an Address using the uint32 representation of an v4 IP
+func V4Address(ip uint32) Address {
+	var a v4Address
+	a[0] = byte(ip)
+	a[1] = byte(ip >> 8)
+	a[2] = byte(ip >> 16)
+	a[3] = byte(ip >> 24)
+	return a
+}
+
+// V4AddressFromString creates an Address using the string representation of an v4 IP
+func V4AddressFromString(ip string) Address {
+	var a v4Address
+	copy(a[:], []byte(net.ParseIP(ip))[12:16])
```

Comment:
You can avoid the cast here by just addressing the `ip` slice directly since it is a `[]byte`.